packages/core/src/tools/error.ts

```diff
@@ -110,3 +137,20 @@ export function createHandlingStack(): string {
 
   return formattedStack!
 }
+
+export function flattenErrorCauses(error: RawError, parentSource: ErrorSource): ErrorCause[] {
+  let count = 0
+  let currentError = error
+  const causes: Array<RawError | ErrorCause> = []
+  while (currentError?.cause && currentError.cause && count < 10) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏replace `count < 10` with `causes.length < 10`